### PR TITLE
BulkWriteOperation.execute() now throws BulkWriteException

### PR DESCRIPTION
**Pull request by MustafaHaddara**
Fixes https://github.com/fakemongo/fongo/issues/328

tl;dr: In mongo-java-driver 3.4.2 (and higher), calling `BulkWriteOperation.execute()` throws a `BulkWriteException` when some of the writes fail. fongo was throwing `DuplicateKeyException` for the same operation. This PR brings fongo in line with that behavior. 

I left a comment in [translateWriteErrorsToNew](https://github.com/fakemongo/fongo/pull/334/files#diff-754e00c5766f1acd0423113c799a39f4R1612): the tests in `FongoInsertManyTest`, such as `should_throw_MongoBulkWriteException_if_middle_document_is_duplicated_and_ordered`, [specifically check for an _empty_ document in the `details` field](https://github.com/fakemongo/fongo/blob/master/src/test/java/com/github/fakemongo/FongoInsertManyTest.java#L75) of the `BulkWriteError`. 

This feels weird to me because it means I have to throw away data when I translate a `com.mongodb.BulkWriteError` to a `com.mongodb.bulk.BulkWriteError`, but it's what the unit tests are looking for and I'm assuming those unit tests are validating how mongo behaves, so I maintained that behavior instead of modifying the test.

### CODE REVIEW

**Comment by boweihan** [comment]
What's the status on this?

**Comment by twillouer** [comment]
thanks !
